web/gui/dashboard/src/domains/global/selectors.ts

```diff
@@ -0,0 +1,40 @@
+import { prop } from "ramda"
+import { createSelector } from "reselect"
+
+import { AppStateT } from "store/app-state"
+
+import { GetKeyArguments, getKeyForCommonColorsState } from "./reducer"
+import { storeKey } from "./constants"
+
+export const createSelectAssignedColors = (args: GetKeyArguments) => (state: AppStateT) => {
+  const keyName = getKeyForCommonColorsState(args)
+  const substate = state[storeKey].commonColorsKeys[keyName]
```

Comment:
For variable-dependant selection and &&-based protection just the R.path could be used, I think.